### ISSUE TITLE
[TTNN] Add ability to workaround ops that requires tensors in specific dimension

### ISSUE THREAD

**Issue by sdjordjevicTT** [tenstorrent/tt-mlir#1517]
There are examples in TTNN where certain ops can only execute on 4D tensors. Add a possibility to workaround such ops by inserting reshapes before and after.

**Comment by skhatriTT** [comment]
@sdjordjevicTT is this still active/needed?

**Comment by sdjordjevicTT** [comment]
Yep, this is still active and nice to have, not a priority right now, marked as P3 for now...
